packages/terra-application-layout/src/ApplicationLayout.jsx

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AppDelegate from 'terra-app-delegate';
+import NavigationLayout from 'terra-navigation-layout';
+import { routeConfigPropType } from 'terra-navigation-layout/lib/configurationPropTypes';
+import { matchPath } from 'react-router-dom';
+import { withModalManager } from 'terra-modal-manager';
+import { ApplicationLinksPropType } from 'terra-application-links';
+
+import RoutingMenu from './menu/RoutingMenu';
+import ApplicationMenuWrapper from './menu/_ApplicationMenuWrapper';
+import ApplicationHeader from './header/_ApplicationHeader';
+import ApplicationLayoutPropTypes from './utils/propTypes';
+import Helpers from './utils/helpers';
+import UtilityHelpers from './utils/utilityHelpers';
+
+const navigationLayoutSizes = ['default', 'tiny', 'small', 'medium', 'large', 'huge'];
+
+const propTypes = {
+  /**
+   * The AppDelegate instance provided by `withModalManager`. If an AppDelegate instance is
+   * explicitly provided to the ApplicationLayout, the ModalManager will wrap it and
+   * fall back to its defined APIs as needed.
+   */
+  app: AppDelegate.propType,
+  /**
+   * The content to be rendered in the ApplicationLayout's extensions region. This component will be provided an AppDelegate (as `app`) and
+   * a `layoutConfig` as props to facilitate communication with the ApplicationLayout.
+   */
+  extensions: PropTypes.element,
+  /**
+   * The index, or default, path of the routing configuration. The ApplicationLayout will redirect to this path
+   * when the router reaches an unknown location.
+   */
+  indexPath: PropTypes.string.isRequired,
+  /**
+   * The configuration values for the ApplicationName component.
+   */
+  nameConfig: ApplicationLayoutPropTypes.nameConfigPropType,
+  /**
+   * Alignment of the header's navigation primary tabs. ( e.g start, center, end )
+   */
+  navigationAlignment: ApplicationLinksPropType.alignment,
+  /**
+   * An array of Objects describing the ApplicationLayout's primary navigation items.
+   */
+  navigationItems: ApplicationLayoutPropTypes.navigationItemsPropType,
+  /**
+   * The routing configuration Object. This is very similar to the routingConfig supported by the NavigationLayout; however,
+   * the ApplicationLayout only supports configuration for the `menu` and `content` regions of the layout. The '/' path is also blacklisted
+   * within this configuration object, as the ApplicationLayout uses it for navigation purposes. Any configuration provided for the '/' path
+   * will be disregarded.
+   */
+  routingConfig: PropTypes.shape({
```

Comment:
I think we want to update this to be 
```
routingConfig: PropTypes.shape({
   menu: routeConfigPropType.isRequired,
   content: routeConfigPropType.isRequired,
})
```
otherwise below if menu config is not provided, this [buildMenuNavigationItems](https://github.com/cerner/terra-framework/pull/116/files#diff-130c239375985c7c5776d9779d26f7bbR79) will fail in attempting to loop `undefined`

Reply:
There is a presence check in buildMenuNavigationItems. I can add one to buildApplicationMenus, but right now it's always called with a valid value.